l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -811,10 +811,28 @@ def generate_attach_report(self, inv):
         action_report = self.env[action_report_model] \
             .browse(action_report_id)
         report_model = self.env['report']
-        report_model.get_pdf(inv.ids, action_report.report_name)
-        attachment = report_model._attachment_stored(
-            inv, action_report)[inv.id]
+        attachment_model = self.env['ir.attachment']
+        # Generate the PDF: if report_action.attachment is set
+        # they will be automatically attached to the invoice,
+        # otherwise use res to build a new attachment
+        res = report_model.get_pdf(
+            inv.ids, action_report.report_name)
+        if action_report.attachment:
+            attachment = report_model._attachment_stored(
+                inv, action_report)[inv.id]
+        else:
+            filename = inv.number
+            data_attach = {
+                'name': filename,
+                'datas': base64.b64encode(res),
+                'datas_fname': filename,
+                'type': 'binary'
+            }
+            attachment = attachment_model.create(data_attach)
```

Comment:
@SimoRubi please add a comment explaining these `if` `else`

Reply:
@eLBati done in https://github.com/OCA/l10n-italy/pull/643/commits/4b57d33961e42edb69028474e59c315116ab9663